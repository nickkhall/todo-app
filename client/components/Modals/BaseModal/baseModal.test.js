// Import React for <div /> elem in mockProps
import React from 'react';

// Material UI Components
import Dialog from '@material-ui/core/Dialog';

// Utils
// @TODO: This needs to be extracted out into its own JEST module
import {
  testingFunc,
  testCommonComponentAttrs,
  getComponentWrapper
} from 'utils/tests';

// Components
import Modal from './baseModal';

// Mock functions
const testCloseModal = testingFunc('onClose');
const mockCloseModalFn = jest.fn();

describe('<Modal /> component', () => {
  const mockProps = {
    title: 'Mock Title',
    children: <div>Hello Modal!</div>,
    onClose: mockCloseModalFn
  };

  const wrapper = getComponentWrapper(Modal, mockProps);

  testCommonComponentAttrs(Modal, mockProps);

  it('should render all components properly', () => {
    expect(wrapper.find(Dialog)).toHaveLength(1);
  });

  it(testCloseModal, () => {
    expect(mockCloseModalFn).not.toHaveBeenCalled();
    const wrapperInstance = wrapper.instance();
    wrapperInstance.props.onClose();
    expect(mockCloseModalFn).toHaveBeenCalled();
  });
});
