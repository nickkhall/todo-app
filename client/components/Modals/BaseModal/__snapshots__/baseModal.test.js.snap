// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Modal /> component when rendering should render a component matching the snapshot 1`] = `
<WithStyles(Dialog)
  aria-describedby="alert-dialog-description"
  aria-labelledby="alert-dialog-title"
  disableBackdropClick={true}
  fullWidth={true}
  maxWidth="sm"
  onExiting={[MockFunction]}
  open={true}
  scroll="paper"
  style={
    Object {
      "padding": 0,
    }
  }
>
  <WithStyles(DialogTitle)>
    Mock Title
  </WithStyles(DialogTitle)>
  <div>
    Hello Modal!
  </div>
</WithStyles(Dialog)>
`;
